### ISSUE TITLE
Tags are not marked to export

### ISSUE THREAD

**Issue by danielcgold** [safx/LightroomPlugin-ClarifaiTagger#11]
When I export tags added by this plugin, they don't show up in the available "include on export" list.

<img width="478" alt="screen shot 2017-02-27 at 11 59 24 pm" src="https://cloud.githubusercontent.com/assets/1226984/23392331/dfebb2b2-fd48-11e6-9d0b-c01f42dcf1e4.png">

**Comment by LowellMontgomery** [comment]
Sorry about that. I guess that was in there from the start, and although I'd noticed it, I'd thought it was just a Lightroom quirk (using the default from the previously created keyword). I think I had been doing so much keywording that I hadn't really identified it as an issue with this plugin (which was there before I ever touched this work, but of course we want to call the catalog:createKeyword() function with "include on export" default as "true" (what would be the point of doing otherwise?)

If fixed this in a bugfix branch, which I've just pushed, tested that the change works, and will merge my pull request now, so you should see this in the "master" version very shortly.

**Comment by LowellMontgomery** [comment]
@danielcgold Thanks for reporting this issue; I'm sure it will save us all some headaches. :wink:

I'm also fixing this in my fork of the [mcrosson/lr_plugin_computer_vision_tagging](https://github.com/mcrosson/lr_plugin_computer_vision_tagging) project, which is worth taking a look at since it is where I've been innovating more, along with the main author of that (very related) project (which initially started as a fork of this one). It uses an export plugin approach, which means the workflow is a little nicer, and has a number of extra features. I've added all my my good work on this project to that one (the support for keyword hierarchies, auto-select threshold, among other things). That project hasn't seen a lot of action in the past couple of months, either, since we've all be busy with OtherStuff™, but it might be the one to keep an eye on. BTW, are you using Mac or Windows? (If Windows, then I managed to fix the issues that prevented Windows users from using this one... I've still not had someone check; I suspect it is still broken for Windows users, but I did make a small change a while back that *might* have some impact on the issue)

**Comment by danielcgold** [comment]
Amazing! I also updated my catalog database (with SQLite) and set the legacy flags to "include on export".

**Comment by LowellMontgomery** [comment]
Funny... :smile: I did the same thing (after first making a backup of my Lr Catalog, some time back). Actually, it wasn't related to this, but to a special use case that's exactly the opposite. I'd created an import list for a bunch of terms that I didn't want to export, but hadn't properly decided that until they were in use, so I bulk changed a couple thousand keywords using an SQL tool... to "don't export".

**Comment by LowellMontgomery** [comment]
@danielcgold I'm assuming you are on Mac? (Although it would be very cool to get a confirmation that this works now for Windows users—but I think there is probably more to do to get it working for them and if it's still broken for them, I'll probably just stick with working on the other one (which, at first, was broken for Mac users, but is now solidly cross-platform since the main author uses Windows and I've been contributing from a Mac)

**Comment by danielcgold** [comment]
Yep, on on a Mac.

**Comment by danielcgold** [comment]
Also, I don't think this is working yet. I confirmed in my local code that `local keyword = catalog:createKeyword(kwName, {}, true, nil, true)` (so the `includeOnExport` param set to true). I disabled/deleted/reloaded the plugin and the keywords still don't seem to come over in the export.

**Comment by LowellMontgomery** [comment]
Ah... suspected you were on Mac, but was hoping it was working for Windows now, though I don't expect the issue to have been so simple. @mcrosson looked into it and tried to get it working for Windows before he gave up and tried the route he took for the related plugin.

Regarding not seeing keywords in the exported images... that could also be your export settings. At least the "new keywords" created with this, after I made the change, showed up with the appropriate checkbox selected (include on export), so I think that if you have _newly-created_ keywords from this plugin, it _should_ work. Just make sure that your export settings for metadata include keywords. Of course there could be some other Lightroom bug at play, but I do think that this plugin is doing what it should do now, in terms of what the newly-created keywords look like in the catalog.

Of course the change will have no effect on keywords that were created before this fix and adding them, if you haven't already made the changes to those keywords in the catalog, will still result in un-exported keywords. Perhaps you have some that didn't get the change in the SQLite DB? At least in my own test (which did not include actually exporting anything), the couple suggested keywords I added to an image that were not already in my keyword catalog, were created with "include on export" selected.
